src/vm/eventpipeprovider.cpp

```diff
@@ -73,13 +77,36 @@ EventPipeProvider::~EventPipeProvider()
     }
 }
 
+const GUID& EventPipeProvider::GenerateGuidFromName()
```

Comment:
I think we should plumb the GUID through if we need it, rather than re-generating.  That way we avoid bugs where they don't match.